src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1196,13 +1196,18 @@ private bool IsValidOverrideReturnType(Symbol overridingSymbol, TypeWithAnnotati
             }
 
             var conversions = compilation.Conversions.WithNullability(true);
+            var overriddenParameters = overriddenMethod.Parameters;
+            var overridingParameters = overridingMethod.Parameters;
+            var overridingMethodOffset = invokedAsExtensionMethod ? 1 : 0;
+            Debug.Assert(overriddenMethod.ParameterCount == overridingMethod.ParameterCount - overridingMethodOffset);
```

Comment:
nit: It may be worth renaming the parameters, otherwise I found this assertion confusing. Maybe `overriddenMethod`->`targetMethod` and `overridingMethod`->`implementationMethod` or something like that?

Reply:
I am definitely open to improving the naming in a follow-up PR. I find that scanning `overriding` and `overridden` the many times they appear is very hard to get used to. `targetMethod` and `implementationMethod`, or `baseMethod` and `overrideMethod`, etc. or some other pair of terms may be more suitable.